pydantic/aliases.py

```diff
@@ -0,0 +1,125 @@
+        choices: A list containing a string or `AliasPath`.
+    """
+
+    choices: list[str | AliasPath]
+
+    def __init__(self, first_choice: str | AliasPath, *choices: str | AliasPath) -> None:
+        self.choices = [first_choice] + list(choices)
+
+    def convert_to_aliases(self) -> list[list[str | int]]:
+        """Converts arguments to a list of lists containing string or integer aliases.
+
+        Returns:
+            The list of aliases.
+        """
+        aliases: list[list[str | int]] = []
+        for c in self.choices:
+            if isinstance(c, AliasPath):
+                aliases.append(c.convert_to_aliases())
+            else:
+                aliases.append([c])
+        return aliases
+
+
+@dataclasses.dataclass(**_internal_dataclass.slots_true)
+class AliasGenerator:
+    """Usage docs: https://docs.pydantic.dev/2.6/concepts/alias#using-an-aliasgenerator
+
+    A data class used by `alias_generator` as a convenience to create various aliases.
+
+    Attributes:
+        validation_alias: A callable that takes a field name and returns a validation alias for it.
+        alias: A callable that takes a field name and returns an alias for it.
+        serialization_alias: A callable that takes a field name and returns a serialization alias for it.
+    """
+
+    validation_alias: Callable[[str], str | AliasPath | AliasChoices] | None = None
+    alias: Callable[[str], str] | None = None
+    serialization_alias: Callable[[str], str] | None = None
+
+    def __init__(
+        self,
+        alias: Callable[[str], str] | None = None,
+        *,
+        validation_alias: Callable[[str], str | AliasPath | AliasChoices] | None = None,
+        serialization_alias: Callable[[str], str] | None = None,
+    ) -> None:
+        """Initialize the alias generator."""
+        self.validation_alias = validation_alias
+        self.alias = alias
+        self.serialization_alias = serialization_alias
+
+    def _generate_alias(
+        self,
+        alias_kind: Literal['validation_alias', 'serialization_alias', 'alias'],
+        allowed_types: tuple[type[str] | type[AliasPath] | type[AliasChoices], ...],
+        field_name: str,
+    ) -> str | AliasPath | AliasChoices | None:
+        """Generate an alias of the specified kind. Returns None if the alias generator is None.
+
+        Raises:
+            TypeError: If the alias generator produces an invalid type.
+        """
+        alias = None
+        alias_generator = getattr(self, alias_kind)
+        if alias_generator is not None:
+            alias = alias_generator(field_name)
+            if alias and not isinstance(alias, allowed_types):
+                raise TypeError(
+                    f'Invalid `{alias_kind}` type. `{alias_kind} generator` must produce' f' one of `{allowed_types}`'
+                )
+        return alias
+
+    def __call__(self, field_name: str) -> tuple[str | AliasPath | AliasChoices | None, str | None, str | None]:
```

Comment:
I think it would be more explicit and easier to understand if this was a named method, like `generate_alises`.